Automattic/jetpack
# CI: Cancel duplicate workflow runs

#### Changes proposed in this Pull Request:

GH Actions doesn't do anything to prevent wasting time continuing to run workflows that have been obsoleted by later pushes. But you can roll your own mechanism to do this with more actions.

OTOH, this does mean that every time one of the target workflows is triggered, we'll wind up triggering a copy of the "Cancelling" workflow too. I don't think we've worked out yet whether GH's "You're queuing too many workflows" is a rate limit or a queue size limit, if the former this might just make it worse rather than better.

#### Jetpack product discussion


p1610636446030600/1610588708.000700-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Pretty much your only option is to fork the repo with this merged into master, then do a bunch of pushes to various branches and see that things get cancelled or not appropriately. I did that at https://github.com/anomiex/testing/actions, you can see it cancelled some duplicate runs for pushes to "tmp" but didn't for simultaneous pushes to "tmp-7", "tmp-8", and "tmp-9" as intended.

#### Proposed changelog entry for your changes:


* None needed.
